File: Bug_Hunt_Practice/Question 8/solution.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

void printVector(const vector<int>& vec) {
    for (int val : vec) {
        cout << val << " ";
    }
    cout << endl;
}

vector<int> cumulativeLengths(const vector<string>& words) {
    vector<int> lengths;
    int currentSum = 0;
    
    for (int i = 0; i < words.size(); i++) {
        lengths.push_back(words[i].length());
        currentSum += words[i].length();
    }
    return lengths;
}

int main() {
    vector<string> words = {"hello", "world", "bug"};
    
    vector<int> result = cumulativeLengths(words);
    
    cout << "Expected Output: 5 10 13" << endl;
    cout << "Actual Output:   ";
    printVector(result);
    return 0;
}

File: Bug_Hunt_Practice/Question 8/solution.py
def cumulative_lengths(words):
    lengths = []
    current_sum = 0
    
    for word in words:
        lengths.append(len(word))
        current_sum += len(word)
        
    return lengths

# --- Main execution ---
words = ["hello", "world", "bug"]
result = cumulative_lengths(words)

print(f"Expected Output: [5, 10, 13]")
print(f"Actual Output:   {result}")